watcher/src/watchers/MoonbeamWatcher.ts

```diff
@@ -1,27 +1,30 @@
-import { sleep } from '@wormhole-foundation/wormhole-monitor-common';
+import { NETWORK, sleep } from '@wormhole-foundation/wormhole-monitor-common';
 import axios from 'axios';
 import { AXIOS_CONFIG_JSON, RPCS_BY_CHAIN } from '../consts';
 import { EVMWatcher } from './EVMWatcher';
 
 export class MoonbeamWatcher extends EVMWatcher {
-  constructor() {
-    super('moonbeam');
+  constructor(network: NETWORK) {
+    super(network, 'moonbeam');
   }
   async getFinalizedBlockNumber(): Promise<number> {
     const latestBlock = await super.getFinalizedBlockNumber();
     let isBlockFinalized = false;
+    if (!RPCS_BY_CHAIN[this.network].moonbeam) {
+      throw new Error('Moonbeam RPC is not defined!');
+    }
+    const rpc: string = (
+      RPCS_BY_CHAIN[this.network].moonbeam ? RPCS_BY_CHAIN[this.network].moonbeam : ''
+    ) as string;
```

Comment:
If you set the variable first, the type refinement will work :+1:

```suggestion
    const rpc = RPCS_BY_CHAIN[this.network].moonbeam
    if (!rpc) {
      throw new Error('Moonbeam RPC is not defined!');
    }
```